File: custom_scripts/public/js/role.js
frappe.ui.form.on('Role', {
    validate: function(frm) {
        const reserved_roles = ["Administrator", "System Manager"];
        if (reserved_roles.includes(frm.doc.role_name)) {
            frappe.throw(__("Tên Role không hợp lệ, vui lòng chọn tên khác"));
        }
    },
    role_name: function(frm) {
        const reserved_roles = ["Administrator", "System Manager"];
        if (reserved_roles.includes(frm.doc.role_name)) {
            frappe.msgprint(__("Tên Role không hợp lệ, vui lòng chọn tên khác"));
            frm.set_value("role_name", ""); // Xoá giá trị nhập
        }
    }
});

File: custom_scripts/overrides/role.py
import frappe
from frappe.model.document import Document
from frappe import _

class Role(Document):
	def autoname(self):
		self.name = self.role_name.strip()

	def validate(self):
		reserved_roles = ["Administrator", "System Manager"]
		if self.is_new() and self.name in reserved_roles:
			frappe.throw(_("Tên Role không hợp lệ, vui lòng chọn tên khác"))
